build/common.sh

```diff
@@ -477,49 +506,48 @@ EOF
   }
 }
 
-function kube::build::clean_image() {
-  local -r image=$1
-
-  kube::log::status "Deleting docker image ${image}"
-  "${DOCKER[@]}" rmi ${image} 2> /dev/null || true
-}
-
-function kube::build::clean_images() {
-  kube::build::has_docker || return 0
-
-  kube::build::clean_image "${KUBE_BUILD_IMAGE}"
-
-  kube::log::status "Cleaning all other untagged docker images"
-  "${DOCKER[@]}" rmi $("${DOCKER[@]}" images -q --filter 'dangling=true') 2> /dev/null || true
-}
-
 function kube::build::ensure_data_container() {
   # If the data container exists AND exited successfully, we can use it.
   # Otherwise nuke it and start over.
   local ret=0
   local code=$(docker inspect \
       -f '{{.State.ExitCode}}' \
-      "${KUBE_BUILD_DATA_CONTAINER_NAME}" 2>/dev/null || ret=$?)
+      "${KUBE_DATA_CONTAINER_NAME}" 2>/dev/null || ret=$?)
   if [[ "${ret}" == 0 && "${code}" != 0 ]]; then
-    kube::build::destroy_container "${KUBE_BUILD_DATA_CONTAINER_NAME}"
+    kube::build::destroy_container "${KUBE_DATA_CONTAINER_NAME}"
     ret=1
   fi
   if [[ "${ret}" != 0 ]]; then
-    kube::log::status "Creating data container ${KUBE_BUILD_DATA_CONTAINER_NAME}"
+    kube::log::status "Creating data container ${KUBE_DATA_CONTAINER_NAME}"
     # We have to ensure the directory exists, or else the docker run will
     # create it as root.
     mkdir -p "${LOCAL_OUTPUT_GOPATH}"
     # We want this to run as root to be able to chown, so non-root users can
     # later use the result as a data container.  This run both creates the data
     # container and chowns the GOPATH.
+    #
+    # The data container creates volumes for all of the directories that store
+    # intermediates for the Go build. This enables incremental builds across
+    # Docker sessions. The *_cgo paths are re-compiled versions of the go std
```

Comment:
wonder if there's some way we can explode this list based on `KUBE_CLIENT_PLATFORMS` or `KUBE_CROSSPLATFORMS` or whatever. (maybe in a follow-up?)

also, we didn't do this previously - why is it necessary now? just to improve performance?

and when are these directories created?